kinisi/parser.py

```diff
@@ -347,3 +361,77 @@ def get_indices(
+
+    :param structure: Initial structure.
+    :param coords: fractional coordinates for all atoms.
+    :param indices: indices for the atoms in the molecules in the trajectory used in the calculation 
+    of the diffusion.
+    :param masses: Masses associated with indices in indices.
+    :param framework_indices: Indices of framework to be used in drift correction. If set to None will return all indices that are not in indices.
+
+
+    :return: Tuple containing: Tuple containing: fractional coordinates for centers and framework atoms
+    and Tuple containing: indices for centers used in the calculation 
+    of the diffusion and indices of framework atoms.
+    """
+    drift_indices = []
+    try:
+        indices.values = indices.values - 1
+    except:
+        raise ValueError('Molecules must be of same length')
+
+    n_molecules = len(indices.dims)
+
+    # Removed method for framework_indices
+    for i, site in enumerate(structure):
+        if i not in indices.values:
+            drift_indices.append(i)
+
+    if masses == None:
+        weights = None
+    elif len(masses.values) != indices.values.shape[1]:
+        raise ValueError('Masses must be the same length as a molecule')
+    else:
+        weights = masses.copy()
+
+    coord_array = coords.values
+    new_s_coords = _calculate_centers_of_mass(coord_array, weights.values, indices.values)
+
+    new_coords = sc.array(dims=['time', 'atom', 'dimension'],
+                          values=np.concatenate((new_s_coords, coord_array[:, drift_indices]), axis=1),
+                          unit=sc.units.dimensionless)
+    new_indices = sc.Variable(dims=['atom'], values=list(range(n_molecules)))
+    new_drift_indices = sc.Variable(dims=['atom'], values=list(range(n_molecules, n_molecules + len(drift_indices))))
+
+    return new_coords, new_indices, new_drift_indices
+
+
+def _calculate_centers_of_mass(coords: np.ndarray, weights: np.ndarray, indices: np.ndarray) -> np.ndarray:
+    """
+    Calculates the weighted molecular centre of mass based on chosen weights and indices as per https://doi.org/10.1080/2151237X.2008.10129266
+    The method involves projection of the each coordinate onto a circle to allow for efficient COM calculation
+    
+     :param coords: array of coordinates
+     :param weights: 1D array of weights of elements within molecule
+     :param indices: N by M dimensional array of indices of N molecules of M atoms
+
+     :return: Array containing coordinates of centres of mass of molecules
+    """
+
+    s_coords = coords[:, indices]
+    theta = s_coords * (2 * np.pi)
+    xi = np.cos(theta)
+    zeta = np.sin(theta)
+    xi_bar = np.average(xi, axis=-2, weights=weights)
+    zeta_bar = np.average(zeta, axis=-2, weights=weights)
+    theta_bar = np.arctan2(zeta_bar, xi_bar)
+    new_s_coords = theta_bar / (2 * np.pi)
```

Comment:
If we are keeping this non-scipp for now, let's put something in the TODO-scipp.md file.